src/Microsoft.Diagnostics.Monitoring.EventPipe/Logs/LogObject.cs

```diff
@@ -13,12 +15,21 @@ public class LogObject : IReadOnlyList<KeyValuePair<string, object>>, IStateWith
         public static readonly Func<object, Exception, string> Callback = (state, exception) => ((LogObject)state).ToString();
 
         private readonly string _formattedMessage;
-        private List<KeyValuePair<string, object>> _items = new();
+        private KeyValuePair<string, object>[] _items = new KeyValuePair<string, object>[8];
```

Comment:
Why move from List to array?

Reply:
Would CollectionsMarshal.AsSpan work here?

Reply:
Issue is we only have `netstandard2.1` target here. `CollectionsMarshal` needs something newer. Probably not worth messing with the targets for this?